src/generator/iter.rs

```diff
@@ -22,9 +26,27 @@ pub struct Iter<'a> {
     last_kind: FeEventKind,
 }
 
-impl<'a> Iter<'a> {
+enum IncludeSource<'a> {
+    Resolve,
+    Image(FeInclude<'a>),
+    Synthetic {
+        parameter: FeInclude<'a>,
+        source: Cow<'a, str>,
+    },
+}
+
+impl<'a> MarkdownIter<'a> {
     pub fn new(frontend: Frontend<'a>) -> Self {
-        Iter { frontend: frontend.fuse(), peek: VecDeque::new(), last_kind: FeEventKind::Start }
+        let frontend: Box<FeEvents<'a>> = Box::new(frontend);
+        MarkdownIter { frontend: frontend.fuse(), peek: VecDeque::new(), last_kind: FeEventKind::Start }
+    }
+
+    pub fn with_rustdoc(rustdoc: Rustdoc<'a>) -> Self {
+        let frontend: Box<FeEvents<'a>> = Box::new(rustdoc.map(|event| {
+            WithRange(event, (0..0).into())
```

Comment:
Is it possible to add proper source and span information for diagnostics to be pointing to the original rust source code? I think we should open an issue for that as using the span 0..0 for everything feels weird.